ts/features/bonus/cgn/screens/merchants/CgnMerchantsListByCategory.tsx

```diff
@@ -0,0 +1,161 @@
+import { OnlineMerchant } from "../../../../../../definitions/cgn/merchants/OnlineMerchant";
+import {
+  cgnOfflineMerchantsSelector,
+  cgnOnlineMerchantsSelector
+} from "../../store/reducers/merchants";
+import { Merchant } from "../../../../../../definitions/cgn/merchants/Merchant";
+import { useNavigationContext } from "../../../../../utils/hooks/useOnFocus";
+import CGN_ROUTES from "../../navigation/routes";
+import { getCategorySpecs } from "../../utils/filters";
+import { H1 } from "../../../../../components/core/typography/H1";
+import { IOColors } from "../../../../../components/core/variables/IOColors";
+import GenericErrorComponent from "../../../../../components/screens/GenericErrorComponent";
+import { MerchantsAll } from "./CgnMerchantsListScreen";
+
+const styles = StyleSheet.create({
+  listContainer: {
+    paddingTop: 5,
+    borderTopLeftRadius: 25,
+    borderTopRightRadius: 25,
+    backgroundColor: IOColors.white,
+    top: -20
+  }
+});
+
+const CgnMerchantsListByCategory = () => {
+  const dispatch = useIODispatch();
+  const currentCategory = useIOSelector(cgnSelectedCategorySelector);
+  const onlineMerchants = useIOSelector(cgnOnlineMerchantsSelector);
+  const offlineMerchants = useIOSelector(cgnOfflineMerchantsSelector);
+
+  const categorySpecs = useMemo(
+    () =>
+      fromNullable(currentCategory).fold(undefined, cat =>
+        getCategorySpecs(cat).fold(undefined, c => c)
+      ),
+    [currentCategory]
+  );
+
+  const navigation = useNavigationContext();
+
+  const categoryFilter = currentCategory
+    ? {
+        productCategories: [currentCategory]
+      }
+    : {};
+
+  const initLoadingLists = () => {
+    dispatch(cgnOfflineMerchants.request(categoryFilter));
+    dispatch(cgnOnlineMerchants.request(categoryFilter));
+  };
+
+  React.useEffect(initLoadingLists, [currentCategory]);
+
+  // Mixes online and offline merchants to render on the same list
+  // merchants are sorted by name
+  const merchantsAll = useMemo(() => {
+    const onlineMerchantsValue = getValueOrElse(onlineMerchants, []);
+    const offlineMerchantsValue = getValueOrElse(offlineMerchants, []);
+
+    const merchantsAll = [...onlineMerchantsValue, ...offlineMerchantsValue];
+
+    // Removes possible duplicated merchant:
+    // a merchant can be both online and offline, or may have multiple result by offlineMerchant search API
+    const uniquesMerchants = [
+      ...new Map<OfflineMerchant["id"] | OnlineMerchant["id"], MerchantsAll>(
+        merchantsAll.map(m => [m.id, m])
+      ).values()
+    ];
+
+    return [...uniquesMerchants].sort((m1: MerchantsAll, m2: MerchantsAll) =>
+      m1.name.localeCompare(m2.name)
+    );
+  }, [onlineMerchants, offlineMerchants]);
+
+  const onItemPress = (id: Merchant["id"]) => {
+    navigation.navigate(CGN_ROUTES.DETAILS.MERCHANTS.DETAIL, {
+      merchantID: id
+    });
+  };
+
+  return (
+    <BaseScreenComponent
+      goBack
+      headerTitle={I18n.t(
```

Comment:
Since we are, eventually, using a dynamic key and we don't have a type protection, what you think about defining a fallback in case of missing string?

Reply:
This is not needed as we seen in https://github.com/pagopa/io-app/pull/3738#discussion_r805758582